src/rgw/store/dbstore/common/dbstore.h

```diff
@@ -872,6 +1268,364 @@ class DB {
+        DB::Object *target;
+        RGWObjState obj_state;
+
+        struct MetaParams {
+          ceph::real_time *mtime;
+          map<std::string, bufferlist>* rmattrs;
+          const bufferlist *data;
+          RGWObjManifest *manifest;
+          const string *ptag;
+          list<rgw_obj_index_key> *remove_objs;
+          ceph::real_time set_mtime;
+          rgw_user owner;
+          RGWObjCategory category;
+          int flags;
+          const char *if_match;
+          const char *if_nomatch;
+          std::optional<uint64_t> olh_epoch;
+          ceph::real_time delete_at;
+          bool canceled;
+          const string *user_data;
+          rgw_zone_set *zones_trace;
+          bool modify_tail;
+          bool completeMultipart;
+          bool appendable;
+
+          MetaParams() : mtime(NULL), rmattrs(NULL), data(NULL), manifest(NULL), ptag(NULL),
+          remove_objs(NULL), category(RGWObjCategory::Main), flags(0),
+          if_match(NULL), if_nomatch(NULL), canceled(false), user_data(nullptr), zones_trace(nullptr),
+          modify_tail(false),  completeMultipart(false), appendable(false) {}
+        } meta;
+
+        explicit Write(DB::Object *_target) : target(_target) {}
+
+        int prepare(const DoutPrefixProvider* dpp);
+        int write_data(const DoutPrefixProvider* dpp,
+                               bufferlist& data, uint64_t ofs);
+        int _do_write_meta(const DoutPrefixProvider *dpp,
+            uint64_t size, uint64_t accounted_size,
+            map<string, bufferlist>& attrs,
+            bool assume_noent, bool modify_tail);
+        int write_meta(const DoutPrefixProvider *dpp, uint64_t size,
+            uint64_t accounted_size, map<string, bufferlist>& attrs);
+      };
+
+      struct Delete {
+        DB::Object *target;
+
+        struct DeleteParams {
+          rgw_user bucket_owner;
+          int versioning_status;
+          ACLOwner obj_owner; /* needed for creation of deletion marker */
+          uint64_t olh_epoch;
+          string marker_version_id;
+          uint32_t bilog_flags;
+          list<rgw_obj_index_key> *remove_objs;
+          ceph::real_time expiration_time;
+          ceph::real_time unmod_since;
+          ceph::real_time mtime; /* for setting delete marker mtime */
+          bool high_precision_time;
+          rgw_zone_set *zones_trace;
+          bool abortmp;
+          uint64_t parts_accounted_size;
+
+          DeleteParams() : versioning_status(0), olh_epoch(0), bilog_flags(0), remove_objs(NULL), high_precision_time(false), zones_trace(nullptr), abortmp(false), parts_accounted_size(0) {}
+        } params;
+
+        struct DeleteResult {
+          bool delete_marker;
+          string version_id;
+
+          DeleteResult() : delete_marker(false) {}
+        } result;
+
+        explicit Delete(DB::Object *_target) : target(_target) {}
+
+        int delete_obj(const DoutPrefixProvider *dpp);
+      };
+
+      struct Stat {
```

Comment:
Are these still used?  StatOp is gone from SAL.

Reply:
Missed it.. addressed now.